src/include/io.h

```diff
@@ -0,0 +1,16 @@
+#ifndef _IO_H
+#define _IO_H 1
+
+static __inline void outb(unsigned char value, unsigned short int port)
```

Comment:
Any advantages of __inline over inline?

Reply:
I believe I was drawing inspiration from Linux kernel docs from way back when, but couldn't justify it without looking it up. Here's a link to the one I remember reading: https://www.kernel.org/doc/local/inline.html , though I believe this might be dated since I am seeing `__inline` as opposed to the mentioned `__always_inline`. The point of placing inline functions in header files still makes sense to me. GCC recommends setting it with `__attribute__` since the keyword `inline` will not be inlined unless optimizations are enabled. Additionally, see the definitions for the `__always_inline` macro:
```
x86_64-linux-gnu/sys/cdefs.h
314:# define __always_inline __inline __attribute__ ((__always_inline__))
341:#  define __extern_inline extern __inline __attribute__ ((__gnu_inline__))
```

Might be worth hijacking the `__always_inline` definition since we are using GCC at the moment. Still curious about trying to track down the `__inline` as it appears to be an MSVC holdover that other compilers now tend to support?